src/opnsense/www/js/opnsense_ui.js

```diff
@@ -809,3 +809,59 @@ $.fn.replaceInputWithSelector = function (data, multiple=false) {
         return $.proxy(that.construct, $(this))();
     });
 }
+
+/**
+ * Processes URL hash to activate a tab and/or perform search or edit actions in a grid.
+ * - Supports hashes with direct actions like "#edit=UUID" or "#search=UUID" without a tab.
+ * - If the hash includes a tab name, & must be used (e.g., "#peers&edit=UUID").
+ *
+ * @param {string} gridSelector - The selector for the grid to target (e.g., '#grid-cert').
+ */
+function handleSearchAndEdit(gridSelector) {
+    let gridLoaded = false;
+
+    const processHash = () => {
+        if (!gridLoaded) return;
+
+        const hash = window.location.hash.slice(1);
+        if (!hash) return;
+
+        const splitIndex = hash.indexOf('&');
+        const tabName = splitIndex !== -1 ? hash.substring(0, splitIndex) : null;
+        const action = splitIndex !== -1 ? hash.substring(splitIndex + 1) : hash;
+
+        if (tabName) {
+            const tabElement = $(`a[href="#${tabName}"]`);
+            if (tabElement.length) {
+                tabElement.tab('show');
+            }
+        }
+
+        if (action) {
+            const [prefix, rawPhrase] = action.includes('=') ? action.split('=') : [null, null];
+            const decodedPhrase = rawPhrase ? decodeURIComponent(rawPhrase.trim()) : null;
+            if (!decodedPhrase) return;
+
+            if (prefix === 'edit') {
+                const editButton = $(`${gridSelector} .command-edit[data-row-id="${decodedPhrase}"]`);
```

Comment:
the edit button may not exist yet, as it is dependent on the front-end having rendered the row, which can only be guaranteed if the `loaded.rs.jquery.bootgrid` triggers again after `trigger('keyup')`.

Perhaps there's a better approach by talking to the bootgrid implementation directly. The edit button fires the bootgrid instance's `show_edit_dialog` function, maybe we can do something like this?

```
function handleSearchAndEdit(gridSelector, getEndpoint) {
let bootgridInstance = gridSelector.data("ui.bootgrid");
bootgridInstance.show_edit_dialog(null, getEndpoint);
}
```

Don't know if this works (.data() might contain a different key or nothing at all). But this allows us to skip any dependency on the front-end and just call the partial template directly.

Reply:
small correction: https://github.com/opnsense/core/blob/master/src/opnsense/www/js/opnsense_bootgrid_plugin.js#L392 is the correct function

Reply:
https://github.com/opnsense/core/pull/8115/commits/3091b3645cb1cf3f94e24fcc97590a617d4008c8